cli/cmd/dashboard.go

```diff
@@ -23,45 +23,35 @@ var dashboardCmd = &cobra.Command{
 			log.Fatalf("port must be positive, was %d", proxyPort)
 		}
 
-		kubectl, err := k8s.NewKubectl(shell.NewUnixShell())
+		api, err := k8s.NewK8sAPI(shell.NewUnixShell().HomeDir(), kubeconfigPath)
 		if err != nil {
-			log.Fatalf("Failed to start kubectl: %v", err)
+			log.Fatalf("NewK8sAPI() failed with: %+v", err)
 		}
 
-		asyncProcessErr := make(chan error, 1)
-
-		err = kubectl.StartProxy(asyncProcessErr, proxyPort)
-
+		url, err := api.ProxyUrlFor(controlPlaneNamespace, "/services/web:http/proxy/", proxyPort)
 		if err != nil {
-			log.Fatalf("Failed to start kubectl proxy: %v", err)
-		}
-
-		url, err := kubectl.UrlFor(controlPlaneNamespace, "/services/web:http/proxy/")
-
-		if err != nil {
-			log.Fatalf("Failed to generate URL for dashboard: %v", err)
+			log.Fatalf("Failed to generate URL for dashboard: %+v", err)
 		}
 
 		fmt.Printf("Opening [%s] in the default browser\n", url)
 		err = browser.OpenURL(url.String())
```

Comment:
Is this opening the browser before starting the proxy? Won't it 404?

Reply:
Yeah it's initially a 404, but Chrome and Safari do a quick reload, so it Just Works.

I think we have a few options:
1) leave it as-is
2) print the URL instead of opening a browser
3) add a delay before opening the browser
4) add some asynchronous signaling around the `StartProxy()` API to let us know when the proxy is ready

2 is the most platform-agnostic but less convenient, 3 is a bit hacky and not reliable, 4 is more complexity. i'm inclined to leave it as-is.

Reply:
@siggy I'm ok with leaving it it it works but maybe we should do `2` anyway just in case the browser open fails?

Reply:
that sounds good. as i write this i'm realizing we already do `2`:
```
$ bin/go-run cli dashboard -p 0
github.com/runconduit/conduit/cli
Opening [http://127.0.0.1:56026/api/v1/namespaces/conduit/services/web:http/proxy/] in the default browser
Starting to serve on 127.0.0.1:56026
```